### PR TITLE
transfer projection to implied subplots when faceting

### PR DESCRIPTION
**Pull request by vnoel**
this catches `transform` kw passed to `plot()` when faceting, and pass the associated projection to the subplots. This does what was suggested in issue #1201.

(sorry for the irrelevant change to reshaping.rst, it seems I've stuck myself in a git hole and can't get out)

### CODE REVIEW

**Comment by fmaussion** [comment]
Thanks! I think we need to document this somewhere, and also some tests would be nice.

Do you feel like trying to add the same functionality to normal 2d plots also? (this will represent a little more work so if you don't have time no big deal).

Also, you should credit yourself in the "what's new" section of ``whats-new.rst``

File: xarray/plot/facetgrid.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ class FacetGrid(object):
 
     def __init__(self, data, col=None, row=None, col_wrap=None,
                  sharex=True, sharey=True, figsize=None, aspect=1, size=3,
-                 subplot_kws=None):
+                 transform=None, subplot_kws=None):
         """
```
Comment (fmaussion) [inline xarray/plot/facetgrid.py]: This new kwarg needs to be documented (see examples just below)

File: xarray/plot/facetgrid.py [hunk 2]
```diff
@@ -142,6 +142,8 @@ def __init__(self, data, col=None, row=None, col_wrap=None,
 
         # Set the subplot kwargs
         subplot_kws = {} if subplot_kws is None else subplot_kws
+        if transform is not None and 'projection' not in subplot_kws:
+            subplot_kws['projection'] = transform
```
Comment (fmaussion) [inline xarray/plot/facetgrid.py]: add: ``# with a transform, we assume the user wants to plot on a projection``

(shorten if this doesn't fit in the pep8 character limit)

**Comment by vnoel** [comment]
One concern I have is that it is conceivable that one might pass a `transform` keyword to `plot()` intended to interact with another library that hijacks matplotlib (i.e. not cartopy). In that case, matplotlib will get an unintended `projection` argument. That is a corner case though.

**Comment by fmaussion** [comment]
> In that case, matplotlib will get an unintended projection argument. That is a corner case though.

Yes, I was about to ask @jhamman and @shoyer about their opinion about this. I'm actually not sure if this really is a corner case...

**Comment by vnoel** [comment]
As it is, the xarray code becomes tied up with cartopy. If cartopy somehow changes its syntax (e.g. by renaming the `transform` kw) this code won't work. 

Maybe there's another, more abstract way to do this?

**Comment by fmaussion** [comment]
> As it is, the xarray code becomes tied up with cartopy. If cartopy somehow changes its syntax (e.g. by renaming the transform kw) this code won't work.

cartopy uses ``transform`` because this is an interface provided by matlotlib: http://matplotlib.org/users/transforms_tutorial.html . This is not likely to change anytime soon.

**Comment by fmaussion** [comment]
I know at least one usage where the ``projection`` keyword is needed for something else than cartopy (for polar plots, see example [here](https://github.com/pydata/xarray/pull/1148#issuecomment-264455218) by @hoonhout), but I don't know how often do people need  a ``transform`` though.

In the end, I'm still in favor of this change:
- I suspect the large majority of the use cases are for cartopy
- it's still possible to override the this default behavior by setting the ``projection`` kwarg yourself

File: xarray/plot/facetgrid.py (hunk 2 above)
Comment (jhamman) [inline xarray/plot/facetgrid.py]: I think we should only make this assumption if the projection is a `cartopy.crs` object. That would be easy to test here.

**Comment by jhamman** [comment]
@vnoel and @fmaussion - my apologies for being quite tardy in looking at this PR. I think this is a valuable change but I do have one comment on the implementation.

**Pull request closed without merging**